plugins/modules/zos_job_submit.py

```diff
@@ -698,13 +699,15 @@ def submit_src_jcl(module, src, src_name=None, timeout=0, hfs=True, volume=None,
                                  "not be cataloged on the volume {1}.".format(src, volume))
                 module.fail_json(**result)
 
-        job_submitted = jobs.submit(src, wait, None, **kwargs)
+        clean_name =  _dsname_escape(src)
+        job_submitted = jobs.submit(clean_name, wait, None, **kwargs)
 
-        # Introducing a sleep to ensure we have the result of job sumbit carrying the job id
+        # Introducing a sleep to ensure we have the result of job submit carrying the job id
         while (job_submitted is None and duration <= timeout):
+            sleep(0.5)
             current_time = timer()
             duration = round(current_time - start_time)
-            sleep(0.5)
+            job_submitted = jobs.submit(clean_name, wait, None, **kwargs)  ### Not sure, but think this is needed
```

Comment:
So apparently this while is just waiting to fail if job_submitted is None. Is it correct that we try to submit a new job?  I don't think so, I would leave as it is since we know big changes are coming to ZOAU 1.3 in terms of job submit and job fetch